src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/UrlRewriteFileParser.cs

```diff
@@ -217,10 +245,14 @@ private void ParseUrlAction(XElement urlAction, UrlRewriteRuleBuilder builder, b
             }
 
             bool appendQuery;
-            if (!bool.TryParse(urlAction.Attribute(RewriteTags.AppendQueryString)?.Value, out appendQuery))
+            if (urlAction.Attribute(RewriteTags.AppendQueryString) == null)
```

Comment:
This code is repetitive, can you extract it to a method?
`var appendQuery = ParseBool(urlAction, RewriteTags.AppendQueryString, default: true);`
or 
`var appendQuery = ParseBool(urlAction, RewriteTags.AppendQueryString) ?? true;`